File: src/main/java/com/itbank/mall/entity/product/Product.java
package com.itbank.mall.entity.product;

import lombok.Data;
import java.time.LocalDateTime;

@Data
public class Product {
    private Long productId;
    private String name;
    private int price;
    private int stock;
    private Long categoryId;
    private String description;
    private String thumbnailUrl;
    private Integer viewCount;
    private Integer recommendCount;
    private Integer notRecommendCount;
    private Boolean isTimeSale;
    private Integer timeSalePrice;
    private LocalDateTime createdAt;
   
    // ✅ 추가: 상품 노출 상태
    private String status;

    // ✅ 추가: 노출 여부 판단 메서드
    public boolean isVisible() {
        return "ACTIVE".equalsIgnoreCase(this.status);
    }
}

File: frontend/types/product.ts
export type Product = {
    description: string; // ← 추가
    productId: number;
    name: string;
    price: number;
    thumbnailUrl: string | null;
    isTimeSale?: boolean;
    timeSalePrice?: number;
    isWished?: boolean;
}


export interface ProductBuyProps {
    productId: number;
    quantity?: number;
    mode?: string | null;
}